In [3]:
import pandas as pd
df = pd.read_excel('Raisin_Dataset.xlsx')
df.sample(5)

Area  MajorAxisLength  MinorAxisLength  Eccentricity  ConvexArea  \
644  121757       486.521068       323.351730      0.747181      125067   
452   95546       487.178282       251.960243      0.855875       99166   
91    56469       325.341425       225.115024      0.721960       58697   
156   45962       251.133384       235.368076      0.348730       47173   
851   73699       408.585555       231.142084      0.824602       76106   

       Extent  Perimeter    Class  
644  0.759913   1357.888    Besni  
452  0.722782   1276.807    Besni  
91   0.685736    933.896  Kecimen  
156  0.742280    810.195  Kecimen  
851  0.703181   1079.578    Besni

In [4]:
X = df[["Area", "MajorAxisLength", "MinorAxisLength", "Eccentricity", "ConvexArea", "Extent", "Perimeter"]]
y = df["Class"]

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=10)

In [5]:
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()
scaler.fit(X_train)

X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [6]:
# No scaling
from sklearn.svm import SVC

model = SVC(kernel="rbf")
model.fit(X_train, y_train)

y_pred = model.predict(X_test)

from sklearn.metrics import classification_report

report = classification_report(y_test, y_pred)
print(report)

model.n_iter_

              precision    recall  f1-score   support

       Besni       0.86      0.75      0.80        83
     Kecimen       0.81      0.90      0.85        97

    accuracy                           0.83       180
   macro avg       0.83      0.82      0.82       180
weighted avg       0.83      0.83      0.83       180



array([229], dtype=int32)

In [7]:
# Scaling Done
from sklearn.svm import SVC

model = SVC(kernel="rbf")
model.fit(X_train_scaled, y_train)

y_pred = model.predict(X_test_scaled)

from sklearn.metrics import classification_report

report = classification_report(y_test, y_pred)
print(report)

model.n_iter_

              precision    recall  f1-score   support

       Besni       0.91      0.83      0.87        83
     Kecimen       0.87      0.93      0.90        97

    accuracy                           0.88       180
   macro avg       0.89      0.88      0.88       180
weighted avg       0.88      0.88      0.88       180



array([382], dtype=int32)

In [11]:
# Linear kernel SVM with no scaling
from sklearn.svm import SVC

model = SVC(kernel="linear")
model.fit(X_train, y_train)

y_pred = model.predict(X_test)

from sklearn.metrics import classification_report

report = classification_report(y_test, y_pred)
print(report)

model.n_iter_

              precision    recall  f1-score   support

       Besni       0.90      0.87      0.88        83
     Kecimen       0.89      0.92      0.90        97

    accuracy                           0.89       180
   macro avg       0.90      0.89      0.89       180
weighted avg       0.89      0.89      0.89       180



array([60953539], dtype=int32)

In [12]:
#from sklearn.svm import SVC
# With Scaling
model = SVC(kernel="linear")
model.fit(X_train_scaled, y_train)

y_pred = model.predict(X_test_scaled)

from sklearn.metrics import classification_report

report = classification_report(y_test, y_pred)
print(report)

model.n_iter_

              precision    recall  f1-score   support

       Besni       0.90      0.84      0.87        83
     Kecimen       0.87      0.92      0.89        97

    accuracy                           0.88       180
   macro avg       0.88      0.88      0.88       180
weighted avg       0.88      0.88      0.88       180



array([1214], dtype=int32)

<h1>For linear kernal, scaling the data reduced the number of iterations drastically (down to 1214 from 85005907). The model performance on the other hand is more or less the same. There is some difference but nothing major.